sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/TransformWithStateExec.scala

```diff
@@ -533,6 +537,16 @@ case class TransformWithStateExec(
     }
   }
 
+  private def getDeserializers(): (InternalRow => Any, InternalRow => Any) = {
+    val getKeyObj =
+      ObjectOperator.deserializeRowToObject(keyDeserializer, groupingAttributes)
+
+    val getValueObj =
+      ObjectOperator.deserializeRowToObject(valueDeserializer, dataAttributes)
+
+    (getKeyObj, getValueObj)
+  }
```

Comment:
Is there a reason why we have a lambda here instead of making `getKeyObj` and `getValueObj` `lazy val`s that we have in the constructor of the physical operator? I _think_ it's more canonical to do it that (like the `lazy val`s in `WatermarkSupport`).

An added benefit to using lazy values is that you we don't need to duplicate code (or comments) between the `hasInitialState` and `!hasInitialState` branches.

Reply:
done